src/consensus/validation.h

```diff
@@ -45,9 +45,9 @@ class CValidationState {
         strRejectReason = strRejectReasonIn;
         corruptionPossible = corruptionIn;
         strDebugMessage = strDebugMessageIn;
+        nDoS = level;
         if (mode == MODE_ERROR)
             return ret;
-        nDoS += level;
```

Comment:
What was the original idea behind bumping this? Was it to continue validating and perhaps find additional errors before giving up, to see if the originating peer needs stronger punishment?

Reply:
I don't know the history here, but that seems like a reasonable guess.